### PR TITLE
Role

### PR DESCRIPTION
**Pull request by abhishekanalytics**

### CODE REVIEW

File: app/auth.py [hunk 1]
```diff
@@ -1,115 +1,39 @@
...
+        else:
+            return jsonify(error="Unauthorized")
+    return decorated_function
+
+
+def empolyee_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        current_user_id = get_jwt_identity()
+        current_user = User.get(current_user_id)
```
Comment (aayush-kumar-codes) [inline app/auth.py:34]: admin should be permitted for all functions

File: app/config.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Config:                           
     MONGO_URI = "mongodb://localhost:27017/userdata"
+    SECRET_KEY = 'hello'
```
Comment (pythexcel) [inline app/config.py:3]: Again no use of environment variables

File: app/views/tasks_views.py [hunk 3]
```diff
@@ -1,48 +1,51 @@
...
-        try:
-            data = request.get_json()
-            result = update_task(task_id, description=data.get('description'))
+@tasks_blueprint.route('task/<string:task_id>', methods=["GET", "PUT", "DELETE"])
+@jwt_required()
+def manage_task(task_id):
+    current_user_id = get_jwt_identity()
+    if current_user_id:
+        if request.method == "GET":
+            task = get_task_by_id(task_id)
```
Comment (aayush-kumar-codes) [inline app/views/tasks_views.py:36]: You should be getting task_id along with user_id until the role is employee.
If it's admin, then he can fetch any task.
If it's manager then he only fetch the tasks to which he is the assignee.

File: app/views/user_views.py [hunk 4]
```diff
@@ -39,27 +46,34 @@ def create_users():
...
             )
             return jsonify(result)
         except Exception as e:
             return jsonify(error=f"Error creating user: {e}")
+        
 
-@user_blueprint.route('id/<string:user_id>', methods=["GET", "PUT", "DELETE"])
+@user_blueprint.route('user/<string:user_id>', methods=["GET", "PUT", "DELETE"])
+@jwt_required()
+@manager_required
```
Comment (aayush-kumar-codes) [inline app/views/user_views.py:59]: Why manager is required for managing user details?

File: app/auth.py [hunk 5]
```diff
@@ -1,115 +1,39 @@
...
-
-
-
+from flask_jwt_extended import get_jwt_identity
+from flask import jsonify
+from functools import wraps
+from app.schema.schema import User
+
+def admin_required(f):
+    @wraps(f)
```
Comment (aayush-kumar-codes) [inline app/auth.py:7]: Move these decorators in separate file, Named decorators.py

**Pull request merged**